docs/resource/cloud-provider/aws/ecs.md

```diff
@@ -12,9 +12,10 @@
 | `aws.ecs.cluster.arn` | string | The ARN of an [ECS cluster](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/clusters.html). | `arn:aws:ecs:us-west-2:123456789123:cluster/my-cluster` | Recommended |
 | `aws.ecs.container.arn` | string | The Amazon Resource Name (ARN) of an [ECS container instance](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/ECS_instances.html). | `arn:aws:ecs:us-west-1:123456789123:container/32624152-9086-4f0e-acae-1a75b14fe4d9` | Recommended |
 | `aws.ecs.launchtype` | string | The [launch type](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/launch_types.html) for an ECS task. | `ec2` | Recommended |
-| `aws.ecs.task.arn` | string | The ARN of an [ECS task definition](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task_definitions.html). | `arn:aws:ecs:us-west-1:123456789123:task/10838bed-421f-43ef-870a-f43feacbbb5b` | Recommended |
-| `aws.ecs.task.family` | string | The task definition family this task definition is a member of. | `opentelemetry-family` | Recommended |
-| `aws.ecs.task.revision` | string | The revision for this task definition. | `8`; `26` | Recommended |
+| `aws.ecs.task.arn` | string | The ARN of a running [ECS task](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/ecs-account-settings.html#ecs-resource-ids). | `arn:aws:ecs:us-west-1:123456789123:task/10838bed-421f-43ef-870a-f43feacbbb5b`; `arn:aws:ecs:us-west-1:123456789123:task/my-cluster/task-id/23ebb8ac-c18f-46c6-8bbe-d55d0e37cfbd` | Recommended |
```

Comment:
Can you add the change to `aws.ecs.task.arn` to the changelog?

Would be good to check what other detectors for this semantic convention do (you can see some references here: https://github.com/search?q=org%3Aopen-telemetry%20aws.ecs.task.arn&type=code), to see if the others are also aligned with what the Collector does or the Collector is different here and we need to approach this differently

Reply:
I've updated the changelog, and briefly reviewed some of the 53 files your link shows. The ones I have reviewed, and (I'm assuming) all the rest, incorrectly link to ECS TaskDefinition documentation to explain what `aws.ecs.task.id` is. The issue is that, this attribute is only extracted by `resourcedetectionprocessor`, and that processor takes that value from ECS Metadata Endpoint.

Now, ECS Metadata Endpoint *does not return ECS Task Definition ARN*. It does return the ARN of the ECS Task (i.e. the running thing) and ARN of the Container (i.e.: the executed docker image). You can see the example response of the ECS Metadata Endpoint here: https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task-metadata-endpoint-v4.html#task-metadata-endpoint-v4-examples

I'm sorry, but fixing that doc in all the possible repositories and libraries is beyond my time capabilities. https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/8274 was supposed to be a "good first issue" and now it turns into a multi-headed hydra....

Reply:
The funny thing is, that the comments are quite obviously contradicting. Take for example this one:

https://github.com/open-telemetry/opentelemetry-go/blob/0c1c434c70c83d8d68585f7c510823a06aa318a2/semconv/v1.13.0/resource.go#L199-L206
![image](https://github.com/open-telemetry/semantic-conventions/assets/2553638/9ef89c7a-b443-48e1-9375-9677f03beb98)

It says it's an ECS Task Definition ARN, but then provides an example which looks like ECS Task ARN.
See: https://docs.aws.amazon.com/AmazonECS/latest/userguide/ecs-account-settings.html#ecs-resource-ids

Reply:
Thank you for checking! Based on your messages I am in favor of doing the change you proposed.

> I'm sorry, but fixing that doc in all the possible repositories and libraries is beyond my time capabilities

No need to do this, no worries. We can just update it on this repository and let the individual language libraries handle it on their own.